bundles/io/org.eclipse.smarthome.io.net/src/main/java/org/eclipse/smarthome/io/net/http/HttpUtil.java

```diff
@@ -180,13 +182,19 @@ public static String executeUrl(String httpMethod, String url, Properties httpHe
     private static ContentResponse executeUrlAndGetReponse(String httpMethod, String url, Properties httpHeaders,
             InputStream content, String contentType, int timeout, String proxyHost, Integer proxyPort, String proxyUser,
             String proxyPassword, String nonProxyHosts) throws IOException {
-        startHttpClient(CLIENT);
+
+        // Referenced http client factory not available
+        if (httpClientFactory == null) {
+            throw new IllegalStateException("Http client factory not available");
+        }
+        // Get shared http client from factory "on-demand"
+        final HttpClient httpClient = httpClientFactory.getCommonHttpClient();
 
         HttpProxy proxy = null;
-        // only configure a proxy if a host is provided
+        // Only configure a proxy if a host is provided
         if (StringUtils.isNotBlank(proxyHost) && proxyPort != null && shouldUseProxy(url, nonProxyHosts)) {
-            AuthenticationStore authStore = CLIENT.getAuthenticationStore();
-            ProxyConfiguration proxyConfig = CLIENT.getProxyConfiguration();
+            AuthenticationStore authStore = httpClient.getAuthenticationStore();
+            ProxyConfiguration proxyConfig = httpClient.getProxyConfiguration();
```

Comment:
@maggu2810 @sjka @DaHoC this seems very (thread) unsafe to me. We share a common client but its mutable and we change configuration on the fly which might not be what other consumers expect for that same instance of the client.

There could even be multiple calls (at the same instance) to the HttpUtil requiring different proxy configs, we did expose ` public static String executeUrl(String httpMethod, String url, Properties httpHeaders, InputStream content,
            String contentType, int timeout, String proxyHost, Integer proxyPort, String proxyUser,
            String proxyPassword, String nonProxyHosts)` so it could be used. 

Are we sure we are happy with this setup?

Reply:
@martinvw Hm, no, I don't think we should happy with it. But I don't have a deeper look in the code and I hope the author does.
If a list for the proxies is used and the same instance is removed, I assume we just need to ensure not to run into a CME if we don't use synchronized statements.
The authentication store seems to get increased but never cleaned up again... At least I don't identify it.

Reply:
@martinvw Yes you're right and that's one point I also pointed out (it's not new code anyway).
In general, one network setup most probably uses the same proxy configuration for all these calls for a time, so that the described problem should not arise too often (assumption).

That being said, the code should be as robust as possible imho - we could synchronize just the (access to the) proxy list to eliminate the (however small) possibility of a ConcurrentModificationException.